### ISSUE TITLE
The type of a class's prototype is interpreted as equivalent to the instance type of that class

### ISSUE THREAD

**Issue by jugglinmike** [microsoft/TypeScript#41738]
**TypeScript Version:** v4.2.0-dev.20201130

**Search Terms:** "prototype type", "type of prototype", "typeof prototype"

**Code**

```ts
class C {
    instanceProperty: number;
    constructor() {
        this.instanceProperty = 0;
    }

    classMethod() {}
}

declare let x: C;
declare let y: typeof C.prototype;

x = y;
```

**Expected behavior:** An error like, "Property 'instanceProperty' is missing in type '{ classMethod(): void; }' but required in type 'C'."

**Actual behavior:** No error is reported.

**Playground Link:** https://www.typescriptlang.org/play?target=1&ts=4.2.0-dev.20201130#code/MYGwhgzhAEDC0G8BQ1XQJYDsIBcyeAFMAFAJwHsAHQ0nATwC5pMBXAWwCMaBuFNYcthykWwHOVIAKAJSI+aNDgAW6CADosufETJUa9aAF5oABl4KAvknnRQkCAFlCy8gBMZiK1YFC4AfUoKcXpqIzkFOygnF3dZBAsAGiQrJFdCO1JCaBBnaAAPJlheNIysnJxoRmgQwnIAM39A8mC6al4kPLC6biA

**Related Issues:** none

A more specific case involves referencing properties in type annotations.

```ts
// no error, as expected (since this aligns with how JavaScript resolves
// property references):
declare let a: C['instanceProperty'];

// no error, but expected an error like "Property 'instanceProperty' does not
// exist on type 'C.prototype'":
declare let b: typeof C.prototype.instanceProperty;
```

**Comment by jcalz** [comment]
Duplicate of #11558, #20922, #30516.  

The gist of those issues is that while it would be possible to synthesize a more correct type for a class's prototype, the benefits of doing so would not be worth the big breaking change to existing real-world code.

**Comment by jugglinmike** [comment]
Ah, too bad. Thanks for the references!
